test/unit/com/unifina/service/CommunityJoinRequestServiceSpec.groovy

```diff
@@ -20,6 +24,9 @@ class CommunityJoinRequestServiceSpec extends Specification {
 	final String communityAddress = "0x0000000000000000000000000000000000000000"
 
 	def setup() {
+		service.ethereumService = Mock(EthereumService)
+		service.chain = Mock(MessageChainUtil)
```

Comment:
Should probably use the real `MessageChainUtil` to test that we're using it correctly? (see earlier comment, which discovered a bug not revealed by the tests)

It has no side effects so can easily use the real thing.